configs/pie-components-config/tsconfig.json

```diff
@@ -21,6 +21,8 @@
     "forceConsistentCasingInFileNames": true,
     "allowImportingTsExtensions": true,
     "noEmit": true,
+    "skipLibCheck": true,
```

Comment:
As part of this refactor, I had to upgrade the version of Vite that we're using in the repo.

Unfortunately, there's currently an issue with some of the Vite declarations, that mean when building certain components, the following is output in the console (although the build does still succeed):

```bash
@justeattakeaway/pie-tag:build: [vite:dts] Start generate declaration files...
@justeattakeaway/pie-tag:build: ../../../node_modules/@vitest/utils/dist/error.d.ts:4:58 - error TS2304: Cannot find name 'WeakKey'.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 4 declare function serializeError(val: any, seen?: WeakMap<WeakKey, any>): any;
@justeattakeaway/pie-tag:build:                                                            ~~~~~~~
@justeattakeaway/pie-tag:build: ../../../node_modules/@vitest/utils/dist/error.d.ts:6:96 - error TS2304: Cannot find name 'WeakKey'.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 6 declare function replaceAsymmetricMatcher(actual: any, expected: any, actualReplaced?: WeakSet<WeakKey>, expectedReplaced?: WeakSet<WeakKey>): {
@justeattakeaway/pie-tag:build:                                                                                                  ~~~~~~~
@justeattakeaway/pie-tag:build: ../../../node_modules/@vitest/utils/dist/error.d.ts:6:133 - error TS2304: Cannot find name 'WeakKey'.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 6 declare function replaceAsymmetricMatcher(actual: any, expected: any, actualReplaced?: WeakSet<WeakKey>, expectedReplaced?: WeakSet<WeakKey>): {
@justeattakeaway/pie-tag:build:                                                                                                                                       ~~~~~~~
@justeattakeaway/pie-tag:build: ../../../node_modules/vitest/node_modules/vite/dist/node/index.d.ts:6:41 - error TS2307: Cannot find module 'rollup/parseAst' or its corresponding type declarations.
@justeattakeaway/pie-tag:build: 
@justeattakeaway/pie-tag:build: 6 export { parseAst, parseAstAsync } from 'rollup/parseAst';
```

Having `skipLibCheck` set, removes these errors, but I appreciate it's not ideal. As noted in the [Typescript docs](https://www.typescriptlang.org/tsconfig#skipLibCheck), this is less type-safe.

When we run `yarn why vite`, we see a lot of versions being used throughout ther repo:

```
├─ @nuxt/vite-builder@npm:3.4.3
│  └─ vite@npm:4.3.9 (via npm:~4.3.3)
│
├─ @nuxt/vite-builder@npm:3.4.3 [5f25d]
│  └─ vite@npm:4.3.9 [b37cd] (via npm:~4.3.3 [b37cd])
│
├─ @playwright/experimental-ct-react@npm:1.32.3
│  └─ vite@npm:4.5.2 [9b7b3] (via npm:^4.1.1 [9b7b3])
│
├─ pie-monorepo@workspace:.
│  └─ vite@npm:5.0.12 [c9f5e] (via npm:5.0.12 [c9f5e])
│
├─ vite-node@npm:0.30.1
│  └─ vite@npm:4.5.2 [9b7b3] (via npm:^4.1.1 [9b7b3])
│
├─ vite-node@npm:1.2.1
│  └─ vite@npm:5.0.12 [047b3] (via npm:^5.0.0 [047b3])
│
├─ vitest@npm:1.2.1
│  └─ vite@npm:5.0.12 (via npm:^5.0.0)
│
├─ vitest@npm:1.2.1 [c9f5e]
│  └─ vite@npm:5.0.12 [49894] (via npm:^5.0.0 [49894])
│
├─ wc-vanilla@workspace:apps/examples/wc-vanilla
│  └─ vite@npm:4.5.2 [9b7b3] (via npm:^4.1.1 [9b7b3])
│
└─ wc-vue3@workspace:apps/examples/wc-vue3
   └─ vite@npm:4.5.2 [69a92] (via npm:4.5.2 [69a92])
```

I suspect the underlying issue is caused by different packages using different major versions.

More info can be found here, but I think we should leave this enabled for now, and look to turn it off once Vite 5 has matured a bit. This 'skipLibCheck' is also done in a number of other places in the repo, hence why I opted to get around this issue this way for the time being.

Reply:
Mainly the fact our version a year out of date and we had some deprecation warnings. I've reverted this now.